File: forge/internal/templates/builtin/docker-compose-deploy/docker-compose-deploy.cs
using Forge.Helpers;

var cmd = Cmd.Create("__NAME__", "Build and deploy via docker-compose");
var buildFlag = cmd.Flag("build", "Rebuild images before deploying");
var file = cmd.Option("file", "Docker compose file", "__VAR_COMPOSE_FILE__");
var project = cmd.Option("project", "Project name", "__VAR_PROJECT_NAME__");
cmd.Parse();

var baseArgs = $"-f {file.Value} -p {project.Value}";

if (buildFlag.Value)
{
	Log.Info("building images...");
	Exec.Run("docker", $"{baseArgs} compose build");
}

Log.Info("deploying...");
Exec.Run("docker", $"{baseArgs} compose up -d");
Log.Success("deployed successfully");

File: forge/internal/templates/builtin/docker-compose-deploy/docker-compose-deploy.ts
import { command, info, success, exec } from '#helpers';

const cmd = command('__NAME__', 'Build and deploy via docker-compose');
const buildFlag = cmd.flag('build', 'Rebuild images before deploying');
const file = cmd.option('file', 'Docker compose file', '__VAR_COMPOSE_FILE__');
const project = cmd.option('project', 'Project name', '__VAR_PROJECT_NAME__');
cmd.parse();

const baseArgs = ['-f', file.value, '-p', project.value];

if (buildFlag.value) {
	info('building images...');
	await exec('docker', [...baseArgs, 'compose', 'build']);
}

info('deploying...');
await exec('docker', [...baseArgs, 'compose', 'up', '-d']);
success('deployed successfully');
